core/corehttp/gateway_handler.go

```diff
@@ -98,18 +115,14 @@ func (i *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	extensionIndex := strings.LastIndex(path, ".")
-	if extensionIndex != -1 {
-		extension := path[extensionIndex:]
-		mimeType := mime.TypeByExtension(extension)
-		if len(mimeType) > 0 {
-			w.Header().Add("Content-Type", mimeType)
-		}
-	}
+	w.Header().Set("X-IPFS-Path", p)
 
 	dr, err := i.NewDagReader(nd)
 	if err == nil {
-		io.Copy(w, dr)
+		_, name := path.Split(urlPath)
+		// set modtime to a really long time ago, since files are immutable and should stay cached
+		modtime := time.Unix(1, 0)
+		http.ServeContent(w, r, name, modtime, dr)
```

Comment:
this saddens me: 

>```
   Historically, HTTP required the Expires field-value to be no more
   than a year in the future.  While longer freshness lifetimes are no
   longer prohibited, extremely large values have been demonstrated to
   cause problems (e.g., clock overflows due to use of 32-bit integers
   for time values), and many caches will evict a response far sooner
   than that.
```
From: https://tools.ietf.org/html/rfc7234

At least we can use values larger than a year. we could set something like 5 years. that's basically forever in http browser time.